File: scripts/temp-fix-badges.js
const fs = require('fs');
const path = require('path');

const ROOT_DIR = path.join(__dirname, '..');

function findHtmlFiles(dir, fileList = []) {
    const files = fs.readdirSync(dir);
    files.forEach(file => {
        const filePath = path.join(dir, file);
        if (file.startsWith('.') || file === 'node_modules') return;
        const stat = fs.statSync(filePath);
        if (stat.isDirectory()) {
            findHtmlFiles(filePath, fileList);
        } else if (file.endsWith('.html')) {
            fileList.push(filePath);
        }
    });
    return fileList;
}

const htmlFiles = findHtmlFiles(ROOT_DIR);

htmlFiles.forEach(file => {
    let content = fs.readFileSync(file, 'utf8');
    let changed = false;

    // This regex grabs from <div class="product-rank-card... to <div class="product-rank-badge"...
    // and looks forward to data-asin="..." inside the card to pull the ASIN
    // So we match the full card piece until data-asin
    const regex = /<div class="product-rank-card([^>]*)>\s*(<span class="product-badge">([^<]+)<\/span>|<div class="product-rank-badge"([^>]*)>([^<]+)<\/div>)([\s\S]*?)<span class="price-current price-update" data-asin="([^"]+)">/g;

    content = content.replace(regex, (match, cardAttrs, badgeFull, oldBadgeSpan, badgeAttrs, oldBadgeText, between, asin) => {
        // If it's the old <span class="product-badge">
        if (badgeFull.startsWith('<span')) {
            changed = true;
            return `<div class="product-rank-card${cardAttrs}">
          <div class="product-rank-badge" data-asin-badge="${asin}">${oldBadgeSpan}</div>${between}<span class="price-current price-update" data-asin="${asin}">`;
        }

        // If it's <div class="product-rank-badge"...
        if (badgeAttrs && !badgeAttrs.includes('data-asin-badge')) {
            changed = true;
            return `<div class="product-rank-card${cardAttrs}">
          <div class="product-rank-badge" data-asin-badge="${asin}">${oldBadgeText}</div>${between}<span class="price-current price-update" data-asin="${asin}">`;
        }

        return match;
    });

    if (changed) {
        fs.writeFileSync(file, content, 'utf8');
        console.log('Fixed badges in: ' + file);
    }
});

File: scripts/temp-fix-badges.py
import os
import re

for folder, dirs, files in os.walk('.'):
    # Ignore node_modules, .git, etc
    if 'node_modules' in folder or '.git' in folder: continue
    
    for f in files:
        if f.endswith('.html'):
            path = os.path.join(folder, f)
            with open(path, 'r', encoding='utf-8') as file:
                content = file.read()
                
            original_content = content
            
            # This regex looks for product-rank-card block, and extracts the ASIN from price-current
            # Then it finds product-rank-badge inside it and adds data-asin-badge if missing
            
            def replace_card(match):
                card_html = match.group(0)
                asin_match = re.search(r'data-asin="([^"]+)"', card_html)
                if not asin_match: return card_html
                
                asin = asin_match.group(1)
                
                # Check if badge lacks data-asin-badge
                # Can be <span class="product-rank-badge"> or <div class="product-rank-badge">
                # We replace class="product-rank-badge" with class="product-rank-badge" data-asin-badge="ASIN"
                
                if 'data-asin-badge' not in card_html:
                    card_html = re.sub(r'(class="product-rank-badge"[^>]*)>', r'\1 data-asin-badge="' + asin + '">', card_html)
                    
                    # Also replace <span class="product-rank-badge"...> with <div class="product-rank-badge"...> to unify
                    card_html = re.sub(r'<span (class="product-rank-badge".*?)>(.*?)</span>', r'<div \1>\2</div>', card_html)
                    
                else:
                    # Just unify to div if it's currently span
                    card_html = re.sub(r'<span (class="product-rank-badge".*?)>(.*?)</span>', r'<div \1>\2</div>', card_html)
                
                return card_html
            
            content = re.sub(r'<div class="product-rank-card.*?(?=</section>|<div class="product-rank-card)', replace_card, content, flags=re.DOTALL)
            
            if content != original_content:
                with open(path, 'w', encoding='utf-8') as file:
                    file.write(content)
                print(f"Fixed {path}")
